crates/sqllib/src/array.rs

```diff
@@ -775,7 +874,7 @@ where
 
 #[doc(hidden)]
 #[allow(clippy::needless_range_loop)]
-pub fn array_insert__NN<T>(array: Vec<Option<T>>, pos: i32, value: Option<T>) -> Vec<Option<T>>
+pub fn array_insert__NN<T>(array: Array<Option<T>>, pos: i32, value: Option<T>) -> Array<Option<T>>
```

Comment:
Can we replace this complex code with `Vec::insert()` or is the semantics different?